bundle/manifests/cert-manager-operator.clusterserviceversion.yaml

```diff
@@ -583,22 +583,6 @@ spec:
               labels:
                 name: cert-manager-operator
             spec:
-              affinity:
-                nodeAffinity:
-                  requiredDuringSchedulingIgnoredDuringExecution:
-                    nodeSelectorTerms:
-                    - matchExpressions:
-                      - key: kubernetes.io/arch
-                        operator: In
-                        values:
-                        - amd64
-                        - arm64
-                        - ppc64le
-                        - s390x
-                      - key: kubernetes.io/os
-                        operator: In
-                        values:
-                        - linux
```

Comment:
Should we just remove line 595 ~ 597 and keep others?

Reply:
Since most of this customizing can be done via https://docs.openshift.com/container-platform/4.13/nodes/scheduling/nodes-scheduler-node-affinity.html#olm-overriding-operator-pod-affinity_nodes-scheduler-node-affinity. I would keep that as the preference rather than adding logic to the operator. What we can do is add a link to this on our operator docs.

Reply:
Since OpenShift in general support multi-arch and cert-manager only supports AMD64 we should limit the deployment on AMD64 nodes.

Reply:
In future releases, we have planned multi-arch support for cert-manager, hence this change.

Reply:
@swghosh @thejasn @rbo , today in a multi-arch env I tested current PR:
```
$ oc get clusterversion version
NAME      VERSION                                    AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.14.0-0.nightly-multi-2023-09-21-175604   True        False         11h     Cluster version is 4.14.0-0.nightly-multi-2023-09-21-175604

Some worker shows arm64 while others amd64:
$ oc get no -L kubernetes.io/arch
NAME                                                           STATUS   ROLES                  AGE   VERSION           ARCH
xxia-multi-arch-dpvb5-master-0.c.openshift-qe.internal         Ready    control-plane,master   11h   v1.27.4+2c287eb   amd64
xxia-multi-arch-dpvb5-master-1.c.openshift-qe.internal         Ready    control-plane,master   11h   v1.27.4+2c287eb   amd64
xxia-multi-arch-dpvb5-master-2.c.openshift-qe.internal         Ready    control-plane,master   11h   v1.27.4+2c287eb   amd64
xxia-multi-arch-dpvb5-worker-a-additional-lzkzm                Ready    worker                 11h   v1.27.4+2c287eb   arm64
xxia-multi-arch-dpvb5-worker-a-additional-nqwk6                Ready    worker                 11h   v1.27.4+2c287eb   arm64
xxia-multi-arch-dpvb5-worker-a-xcq5m.c.openshift-qe.internal   Ready    worker                 11h   v1.27.4+2c287eb   amd64
xxia-multi-arch-dpvb5-worker-b-5tsx2.c.openshift-qe.internal   Ready    worker                 11h   v1.27.4+2c287eb   amd64
xxia-multi-arch-dpvb5-worker-f-qlbsm.c.openshift-qe.internal   Ready    worker                 11h   v1.27.4+2c287eb   amd64

$ oc create ns cert-manager-operator
$ oc create -f - << EOF
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: cert-manager-operator-og
  namespace: cert-manager-operator
spec:
  targetNamespaces:
  - cert-manager-operator
EOF

$ cat > sub.yaml << EOF
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: openshift-cert-manager-operator
  namespace: cert-manager-operator
spec:
  channel: stable-v1
  installPlanApproval: Automatic
  name: openshift-cert-manager-operator
  source: rh-stage-operators-catalog-iib-414
  sourceNamespace: openshift-marketplace
  config:
    affinity: {}
EOF

Install it:
$ oc create -f sub.yaml
$ oc get deploy -n cert-manager-operator cert-manager-operator-controller-manager -o yaml | grep -i affinity
Empty

There is no nodeAffinity:
$ oc get deploy -n cert-manager-operator cert-manager-operator-controller-manager -o yaml
...
    spec:
      containers:
      - args:
...

$ oc get po -n cert-manager-operator
$ oc get csv -n cert-manager-operator

Then uninstall it:
$ oc delete sub -n cert-manager-operator openshift-cert-manager-operator
subscription.operators.coreos.com "openshift-cert-manager-operator" deleted

$ oc delete csv -n cert-manager-operator cert-manager-operator.v1.12.0
clusterserviceversion.operators.coreos.com "cert-manager-operator.v1.12.0" deleted

I tried above install and uninstall repeatedly. Due to random scheduling, sometimes the install can succeed:
$ oc get po -n cert-manager-operator -o wide
NAME                                                        READY   STATUS    RESTARTS   AGE     IP            NODE                                                           NOMINATED NODE   READINESS GATES
cert-manager-operator-controller-manager-5dfd5bd76d-cc9td   2/2     Running   0          3h55m   10.128.2.15   xxia-multi-arch-dpvb5-worker-f-qlbsm.c.openshift-qe.internal   <none>           <none>

But sometimes it fails:
$ oc get po -n cert-manager-operator -o wide
NAME                                                        READY   STATUS             RESTARTS   AGE   IP            NODE                                              NOMINATED NODE   READINESS GATES
cert-manager-operator-controller-manager-5dfd5bd76d-xhpnb   1/2     ImagePullBackOff   0          14s   10.131.2.10   xxia-multi-arch-dpvb5-worker-a-additional-lzkzm   <none>           <none>

$ oc get csv -n cert-manager-operator
NAME                            DISPLAY                                       VERSION   REPLACES                        PHASE
cert-manager-operator.v1.12.0   cert-manager Operator for Red Hat OpenShift   1.12.0    cert-manager-operator.v1.11.4   Installing
````

This means:
Without above fix, customers like @rbo can succeed or fail randomly in the multi-arch env.
With above fix, customers like me **still** can succeed or fail randomly in the multi-arch env.
Then why need above fix given with it and without it have same randomness? Why not follow above suggestion to have stable success?

CC @lunarwhite

Reply:
Of course, because you don't have any `nodeSelect` or `nodeAffinity` with `kubernetes.io/arch: amd64` in place. Because the change just remove the whole `nodeAffinity`.... My suggestion was just remove the arm64, ppc64le and s390x from the list not the while nodeAffinity.

Reply:
+1 to @xingxingxia for documenting the multi-arch edge case.
At this point, I'd suggest that we stick to affinity for amd64 and linux OS for the scope of this PR. In the future, when we transition to multi-arch release for 1.13, we can update in follow-up PR to add other architectures.